### PR TITLE
chore(relay): add `FIREZONE_NAME` env variable

### PR DESCRIPTION
**Pull request by thomaseizinger**
Follow-up from #2544.

### CODE REVIEW

File: rust/relay/src/main.rs [hunk 1]
```diff
@@ -60,6 +60,11 @@ struct Args {
     /// If omitted, we won't connect to the portal on startup.
     #[arg(env = "FIREZONE_TOKEN")]
     token: Option<SecretString>,
+    /// Token generated by the portal to authorize websocket connection.
+    ///
+    /// If omitted, we won't connect to the portal on startup.
+    #[arg(env = "FIREZONE_NAME")]
```
Comment (jamilbk) [inline rust/relay/src/main.rs:65]: Does this require a matching terraform update to name the relays for staging / prod?

Comment (thomaseizinger) [reply rust/relay/src/main.rs:65]: It is entirely optional for now so any update to terraform can be made after this PR as well.

Comment (ReactorScram) [reply rust/relay/src/main.rs:65]: Is it a name or a token? (Happened to see this in the PR list and wondered if it was more like the firezone-id or more like the account slug)

Comment (thomaseizinger) [reply rust/relay/src/main.rs:65]: cc @AndrewDryga

File: rust/relay/src/main.rs [hunk 2]
```diff
@@ -60,6 +60,11 @@ struct Args {
     /// If omitted, we won't connect to the portal on startup.
     #[arg(env = "FIREZONE_TOKEN")]
     token: Option<SecretString>,
+    /// Token generated by the portal to authorize websocket connection.
+    ///
+    /// If omitted, we won't connect to the portal on startup.
```
Comment (jamilbk) [inline rust/relay/src/main.rs]:
```suggestion
    /// Used as the human name for this Relay to display in the portal. If not provided,
    /// the system hostname is used by default.
```

**Review by jamilbk — CHANGES_REQUESTED** [review]

File: rust/relay/src/main.rs (hunk 2 above)
Comment (thomaseizinger) [reply rust/relay/src/main.rs]: Ha, copy paste error 😂

**Comment by jamilbk** [comment]
updating title from feat -> chore because we don't (yet) support self-hosting Relays

**Review by jamilbk — APPROVED** [review]

**Pull request merged**